apps/backend/src/schemas/device.py

```
@@ -167,13 +171,24 @@ class DeviceConnectionTest(BaseModel):
     connection_status: str = Field(description="Connection test result")
     response_time_ms: Optional[float] = Field(description="Connection response time")
     error_message: Optional[str] = Field(description="Error message if connection failed")
     test_timestamp: datetime = Field(default_factory=datetime.utcnow)
     
+    @field_validator("ip_address", mode="before")
+    @classmethod
+    def convert_ip_address(cls, v):
+        """Convert IPv4Address/IPv6Address objects to string"""
+        if v is None:
+            return v
+        from ipaddress import IPv4Address, IPv6Address
```

Ruff (0.12.2)

Redefinition of unused `IPv4Address` from line 9

Remove definition: `IPv4Address`

(F811)